zipkin-api.yaml

```diff
@@ -248,17 +248,30 @@ definitions:
     properties:
       traceId:
         type: string
+        description: 'Randomly generated globally unique transaction/context id'
       name:
         type: string
+        description: 'RPC name'
+      parentId:
+        type: string
+        maxLength: 16
+        minLength: 16
+        description: 'Parent span id (16 digit hex value)'
       id:
         type: string
+        maxLength: 16
+        minLength: 16
+        description: 'Id of current span (16 digit hex value)'
       timestamp:
         type: integer
         format: int64
+        description: 'Start time of span as UNIX timestamp in **milliseconds**'
       duration:
         type: integer
+        description: 'Duration of span in **milliseconds**'
       debug:
         type: boolean
+        description: 'Indicator whether this trace was sampled from standard traffic or forced.  Usually coupled to the X-B3-Flags header'
```

Comment:
Thanks for taking the time to explain.  Completely makes sense.  I was coming from a http centric world.  What was also interesting was the comment on parent/root spans where the parentId isn't null but equal to the current span and the traceId.  Seems to be many ways to skin the same cat which makes sense given the number of tracing libraries starting to use zipkin for storage and analysis.